src/libstore/drv-output-info.hh

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "path.hh"
+
+namespace nix {
+
+struct DrvOutputId {
+    StorePath drvPath;
+    std::string outputName;
+
+    std::string to_string() const;
+
+    static DrvOutputId parse(const std::string &);
+
+    bool operator<(const DrvOutputId& other) const { return to_pair() < other.to_pair(); }
+    bool operator==(const DrvOutputId& other) const { return to_pair() == other.to_pair(); }
+
+private:
+    // Just to make comparison operators easier to write
+    std::pair<StorePath, std::string> to_pair() const
+    { return std::make_pair(drvPath, outputName); }
+};
+
+typedef std::variant<StorePath, DrvOutputId> RawDrvInput;
+
+struct DrvInput : RawDrvInput {
+    using RawDrvInput::RawDrvInput;
+
+    std::string to_string() const;
+    static DrvInput parse(const std::string & strRep);
+
+    const RawDrvInput variant() const
+    { return static_cast<RawDrvInput>(*this); }
+};
+
+std::list<std::string> stringify_refs(const std::set<DrvInput> & refs);
```

Comment:
unused

Reply:
Thanks, removed in a2803d672817d2dc1cb3a9992f88232456feb1b8